src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -8360,7 +8360,9 @@ and ComputeMethodAccessRestrictedBySig eenv vspec =
         else
             vspec
 
-    let isHiddenBySignatureVal = IsHiddenVal eenv.sigToImplRemapInfo vspec
+    let isHiddenBySignatureVal =
```

Comment:
This might intervene with the generated `.Is*` members properties which are automatically added for DUs.
I am not sure the problem is real, but the combination of your change and the way I remember the `.Is*` feature rings a bell for me.

Reply:
@T-Gro , yes that popped into my mind while working on this too ... however, the fix for that was to make the compilergenerated Is* methods visible to the TypeChecker.  Here the fix is to ensure that CompilerGenerated methods are not forced to be emitted internal but instead are emitted with the visibility specified when generated for 
generated methods without a sigfile definiton.

Equals for these specified type generated equals is different from other generated Equals methods, because they are overrides, where this is just a new method on the type.  There is a code path that forces the overriden Equals to public, and then the compiler changes it to the visibility of the overriden method definition.

Reply:
@T-Gro , over the weekend I came up with a much better more targeted fix.